versioned_hdf5/wrappers.py

```diff
@@ -377,220 +410,6 @@ def _visit(self, prefix, func):
     # TODO: override other relevant methods here
 
 
-def _make_new_dset(
```

Comment:
All this function was doing was
1. convert data from ArrayLike to a np.ndarray
2. in some edge cases, eagerly convert its dtype to the specified dtgype
3. ...that's it. It was creating a dcpl and discarding it, with no effect. All validation was done in `create_dataset`.